src/types/wrappers/EqualGrowthCommunityModel.jl

```diff
@@ -0,0 +1,101 @@
+"""
+$(TYPEDEF)
+
+A wrapper around [`CommunityModel`](@ref) that returns a community model where
+the growth rates of all members are constrained to be equal to
+`community_objective_id`, which is the community growth rate. The objective of
+the resultant model is set to this `community_objective_id`.
+
+# Notes
+1. No biomass metabolite exists (and none are created).
+
+# Fields
+$(TYPEDFIELDS)
+"""
+Base.@kwdef mutable struct EqualGrowthCommunityModel <: AbstractModelWrapper
+    inner::CommunityModel
+    community_objective_id::String
+end
+
+Accessors.unwrap_model(model::EqualGrowthCommunityModel) = model.inner
+
+Accessors.variables(cm::EqualGrowthCommunityModel) =
+    [variables(cm.inner); cm.community_objective_id]
+
+Accessors.n_variables(cm::EqualGrowthCommunityModel) = n_variables(cm.inner) + 1
+
+Accessors.metabolites(cm::EqualGrowthCommunityModel) =
+    [metabolites(cm.inner); [m.id for m in cm.inner.members]]
+
+Accessors.n_metabolites(cm::EqualGrowthCommunityModel) =
+    n_metabolites(cm.inner) + length(cm.inner.members)
+
+Accessors.balance(cm::EqualGrowthCommunityModel) = [
+    balance(cm.inner)
+    spzeros(length(cm.inner.members))
+]
+
+function Accessors.stoichiometry(cm::EqualGrowthCommunityModel)
+
+    S = stoichiometry(cm.inner)
+    obj_col = spzeros(size(S, 1))
+
+    obj_links = spzeros(length(cm.inner.members), size(S, 2))
+    for i = 1:length(cm.inner.members)
+        m = cm.inner.members[i]
+        j = first(
+            indexin([add_community_prefix(m, m.biomass_reaction_id)], variables(cm.inner)),
```

Comment:
this is quadratic, make a proper lookup table (or use the helper function from semantics to construct the whole matrix)

also, do not update sparse matrices from the middle, it's even more slow

Reply:
what do you mean here?

Reply:
this will have very few updates, just one for each biomass function

Reply:
fixed this